### PR TITLE
common: fix build on Windows

### PR DESCRIPTION
**Pull request by DamianDuy**
---
This change is [Reviewable](https://reviewable.io/reviews/pmem/miniasync/41)

### CODE REVIEW

File: src/miniasync.def [hunk 1]
```diff
@@ -13,3 +13,4 @@ EXPORTS
     vdm_delete
     vdm_descriptor_threads
     vdm_descriptor_threads_polled
+    os_rand_r
```
Comment (wlemkows) [inline src/miniasync.def:16]: This change cannot take place

**Review by wlemkows — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,10 @@
 # Copyright 2021-2022, Intel Corporation
 #
 
+if(MSVC)
+    add_definitions(/FI"platform.h")
```
[the remark this answers is not in the document]
Comment (DamianDuy) [reply src/CMakeLists.txt]: I've deleted this and added platform.h file under #ifdef. I'm not sure if that's what I was supposed to do?

File: src/miniasync.def (hunk 1 above)
Comment (DamianDuy) [reply src/miniasync.def]: I've removed it and added files from the core as a static library, which is then included in the miniasync library and in tests.

**Review by wlemkows — COMMENTED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/41)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @DamianDuy)

**Review by wlemkows — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/41#-:-MvJKrwX7TWsKT0IBO2U:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @DamianDuy)

**Pull request merged**